### PR TITLE
GCI-325 Write tests for ConceptServiceImpl.getConceptNameTagByName

### PR DESCRIPTION
**Pull request by PrathameshMutkure**
## Description of what I changed

I have added tests to cover getConceptNameTagByName method of ConceptServiceImpl class in ConceptServiceImplTest file

## Issue I worked on

see https://issues.openmrs.org/browse/GCI-325

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java [hunk 1]
```diff
@@ -791,4 +791,26 @@ public void getCountOfDrugs_shouldReturnTheTotalNumberOfMatchingNumbers() {
...
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnTheMatchingConceptNameTagObjectIfFound() {
+		ConceptNameTag c = new ConceptNameTag();
+		c.setTag("tag");
+		c.setDescription("some description");
+		assertNull(c.getId());
+		ConceptNameTag savedC = conceptService.saveConceptNameTag(c);
+		assertNotNull(savedC.getId());
+		assertEquals(savedC.getDescription(), conceptService.getConceptNameTagByName("tag").getDescription());
```
Comment (mavrk) [inline api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Two different conceptNameTags can have the same description right? We can just assert that both of the objects are equal.

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Yes, thanks for pointing that out, I'll check for IDs.

Comment (mavrk) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Also, you can write tests based on standardTestDataset.xml which is the test dataset. No need to create a new ConceptNameTag

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java [hunk 2]
```diff
@@ -791,4 +791,26 @@ public void getCountOfDrugs_shouldReturnTheTotalNumberOfMatchingNumbers() {
 		assertEquals(new Integer(1),
 		    conceptService.getCountOfDrugs(phrase, conceptService.getConcept(conceptId), true, true, true));
 	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String)
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnNullIfNoConceptNameTagIsFound() {
+		assertNull(conceptService.getConceptNameTagByName(null));
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: How to do ensure that by requesting `.getConceptNameTagByName(null)` for should return null if No concept nameTag is found?

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: [This](https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java#L873) method here explains that and also [this](https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java#L732) test case shares similar behaviour

Comment (mavrk) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: A 'null' tag will obviously return null for .getConceptNameTagByName(null) How about you try passing a tag value (not null) which is not present in the sampleDataset to see how the test actually works

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Good idea, I'll work on it :)

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java (hunk 1 above)
Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Thanks for suggestion, I'm working on that ;)

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java [hunk 3]
```diff
@@ -65,6 +67,8 @@
 	@Before
 	public void runBeforeAllTests() {
 		conceptService = Context.getConceptService();
+		
+		executeDataSet(STANDARD_TEST_DATASET_XML);
```
Comment (mavrk) [inline api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Isn't this dataset automatically executed?

Comment (mavrk) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: See [this](https://github.com/openmrs/openmrs-core/blob/4a0feb8da351088f25fdc4e6d324a1f277aa3410/api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java#L689) test. It runs on a test [dataset](https://github.com/openmrs/openmrs-core/blob/4a0feb8da351088f25fdc4e6d324a1f277aa3410/api/src/test/resources/org/openmrs/util/databasechange/standardTest-1.9.7-dataSet.xml)

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java [hunk 4]
```diff
@@ -52,6 +52,8 @@
 public class ConceptServiceImplTest extends BaseContextSensitiveTest {
 	
 	protected ConceptService conceptService = null;
+
+	private static final String STANDARD_TEST_DATASET_XML = "org/openmrs/include/standardTestDataset.xml";
```
Comment (mavrk) [inline api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: remove this as well

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: I'll resolve this squashing all commits into 1 :)

**Comment by PrathameshMutkure** [comment]
Thank you @mavrk for suggesting changes, I've resolved all the issues and squashed all commits into 1 :)

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java [hunk 5]
```diff
@@ -791,4 +791,22 @@ public void getCountOfDrugs_shouldReturnTheTotalNumberOfMatchingNumbers() {
...
+	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String) 
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnTheMatchingConceptNameTagObjectIfFound() {
+		ConceptNameTag c = conceptService.getConceptNameTag(1);
+		assertNotNull(c);
+		assertEquals(c.getId(), conceptService.getConceptNameTagByName(c.getTag()).getId());
```
Comment (mavrk) [inline api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Is .getId() really needed? What if we only assert on the objects? The signature of these objects will get matched automatically (id, desc, etc) right?

Comment (mavrk) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: FYI its not important but makes more sense.

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: No worry, I'll do that

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Done!

File: api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java [hunk 6]
```diff
@@ -828,4 +828,22 @@ public void saveConceptProposal_shouldReturnUpdatedConceptProposalObject() {
...
+	public void getConceptNameTagByName_shouldReturnNullIfNoConceptNameTagIsFound() {
+		assertNull(conceptService.getConceptNameTagByName("random-tag"));
+	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String)
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnTheMatchingConceptNameTagObjectIfFound() {
+		ConceptNameTag c = conceptService.getConceptNameTag(1);
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Don't name the variables without proper conventions. Better to rename this `c` to some meaning full name

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Would `conceptNameTag` be fine?

Comment (PrathameshMutkure) [reply api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java]: Looking at naming conventions, I've renamed it :)

**Pull request merged**